disnake/interactions/base.py

```diff
@@ -1033,6 +1041,102 @@ async def autocomplete(self, *, choices: Choices) -> None:
 
         self._responded = True
 
+    @overload
+    async def send_modal(self, modal: Modal) -> None:
+        ...
+
+    @overload
+    async def send_modal(
+        self,
+        *,
+        title: str,
+        custom_id: str,
+        components: Components,
+    ) -> None:
+        ...
+
+    async def send_modal(
+        self,
+        modal: Modal = None,
+        *,
+        title: str = None,
+        custom_id: str = None,
+        components: Components = None,
+    ) -> None:
+        """|coro|
+
+        Responds to this interaction by displaying a modal.
+
+        .. note::
+
+            Not passing a modal here will not register a callback, and a modal_submit interaction
+            will need to be handled manually.
+
+        Parameters
+        ----------
+        modal: :class:`~.ui.Modal`
+            The modal to display. This cannot be mixed with the ``title``, ``custom_id``, and ``components`` parameters.
+        title: :class:`str`
+            The title of the modal. This cannot be mixed with the ``modal`` parameter.
+        custom_id: :class:`str`
+            The ID of the modal that gets received during an interaction.
+            This cannot be mixed with the ``modal`` parameter.
+        components: |components_type|
+            The components to display in the modal. A maximum of 5.
+            This cannot be mixed with the ``modal`` parameter.
+
+        Raises
+        ------
+        TypeError
+            Cannot mix the ``modal`` parameter and the ``title``, ``custom_id``, ``components`` parameters.
+        ValueError
+            Maximum number of components (5) exceeded.
+        HTTPException
+            Displaying the modal failed.
+        ModalInteractionException
+            This interaction cannot be responded with a modal.
+        InteractionResponded
+            This interaction has already been responded to before.
+        """
+        # title and components must always be provided if any of them are provided, but custom_id is optional
+
+        if modal is not None and any((title, components, custom_id)):
+            raise TypeError(f"Cannot mix modal argument and title, custom_id, components arguments")
+
+        parent = self._parent
+        adapter = async_context.get()
+
+        if parent.type is InteractionType.modal_submit:
+            raise ModalInteractionException(parent)  # type: ignore
+
+        if self._responded:
+            raise InteractionResponded(parent)
+
+        modal_data: ModalPayload
+
+        if modal is not None:
+            modal_data = modal.to_components()
+        elif title and components:
+            modal_data = {
+                "title": title,
+                "custom_id": os.urandom(16).hex() if custom_id is None else custom_id,
```

Comment:
Why is this necessary? People need to define a custom_id in order to use `bot.wait_for("modal_submit")`

Reply:
001f5e462b807a75c953b52bc516cf32baf23499